.github/workflows/ci-build.yaml

```diff
@@ -201,12 +210,24 @@ jobs:
           - test: test-executor
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
+          - test: test-executor
+            install_k3s_version: v1.25.11-k3s1
+            profile: minimal
+            pod-status-finalizer: true
+          - test: test-corefunctional
+            install_k3s_version: v1.25.11-k3s1
+            profile: minimal
           - test: test-corefunctional
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
+            pod-status-finalizer: true
           - test: test-functional
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
+          - test: test-functional
+            install_k3s_version: v1.25.11-k3s1
+            profile: minimal
+            pod-status-finalizer: true
```

Comment:
@agilgur5 wanted to bring to your attention that this PR will double our e2e tests I believe. This seems like a good thing for being able to test with the environment variable feature flag both on and off, but thought I'd bring it up to you in case you have any concerns?

Reply:
I wonder if we may want to increase [this](https://github.com/argoproj/argo-workflows/blob/main/.github/workflows/ci-build.yaml#L173) timeout-minutes field at least. Maybe we should double it?

Reply:
> I wonder if we may want to increase [this](https://github.com/argoproj/argo-workflows/blob/main/.github/workflows/ci-build.yaml#L173) timeout-minutes field at least. Maybe we should double it?

The `timeout-minutes` actually applies to _each_ job in the matrix -- each of them has 30min. So unless this feature makes those take substantially longer, I wouldn't increase it.
When I was working on auto-retrying CI (I still have a branch with that WIP), I actually added `timeout-minutes` to each individual job in the matrix as some _shouldn't_ use the full 30 min (i.e. reducing timeouts rather than increasing them).

Reply:
> @agilgur5 wanted to bring to your attention that this PR will increase our e2e tests. This seems like a good thing for being able to test with the environment variable feature flag both on and off, but thought I'd bring it up to you in case you have any concerns?

Yea I am a bit concerned by that. In particular:
1. is it really necessary to run _all_ of the E2E test suites with this flag? or does this feature only apply to some of them? I know the E2E test suite organization has been a bit of a black box without Alex, so that may not be easy to answer 😅 
1. we need to draw a boundary as to which feature flags are tested and which ones aren't. generically speaking, a common approach is to run tests with _all_ flags off (or at their defaults, rather) and with all flags on. selective testing is very complicated to manage otherwise and can exponentially increase the test time to an unreasonable amount.
    1. do we _always_ want to run E2E tests on this feature in particular? since this doesn't just run them for this PR, it runs them for all future commits / PRs.

Reply:
Thanks for the input @agilgur5. I know @sakai-ast was feeling this feature was risky (hence the feature flag) and that was the motivation - that this would catch issues of pods not shutting down. 

But now that you mention it, I wonder if it can give us a bunch of confidence simply seeing these pass here in this PR that maybe we don't need to run for every e2e test.? Maybe we could just add it to `test-functional` or something? What do you think @sakai-ast ?

Reply:
We could also turn it on by default for the E2E tests instead of running it a second time.

Reply:
Thank you for the discussion.
Turning it on by default for the E2E tests is the best safe direction if it can be done.
Because I believe Argo Workflows implementations after this PR is merged will need to be based on this feature, and the feature flag will be removed in the future.
It should be fixed as follows.

```diff
--- a/.github/workflows/ci-build.yaml
+++ b/.github/workflows/ci-build.yaml
@@ -180,19 +180,10 @@ jobs:
         include:
           - test: test-executor
             profile: minimal
-          - test: test-executor
-            profile: minimal
-            pod-status-finalizer: true
-          - test: test-corefunctional
-            profile: minimal
           - test: test-corefunctional
             profile: minimal
-            pod-status-finalizer: true
           - test: test-functional
             profile: minimal
-          - test: test-functional
-            profile: minimal
-            pod-status-finalizer: true
           - test: test-api
             profile: mysql
           - test: test-cli
@@ -210,24 +201,12 @@ jobs:
           - test: test-executor
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
-          - test: test-executor
-            install_k3s_version: v1.25.11-k3s1
-            profile: minimal
-            pod-status-finalizer: true
-          - test: test-corefunctional
-            install_k3s_version: v1.25.11-k3s1
-            profile: minimal
           - test: test-corefunctional
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
-            pod-status-finalizer: true
-          - test: test-functional
-            install_k3s_version: v1.25.11-k3s1
-            profile: minimal
           - test: test-functional
             install_k3s_version: v1.25.11-k3s1
             profile: minimal
-            pod-status-finalizer: true
     steps:
       - name: Install socat (needed by Kubernetes v1.25)
         run: sudo apt-get -y install socat
@@ -291,7 +270,7 @@ jobs:
             LOG_LEVEL=info \
             API=${{matrix.test == 'test-api' || matrix.test == 'test-cli' || matrix.test == 'test-java-sdk' || matrix.test == 'test-python-sdk'}} \
             UI=false \
-            POD_STATUS_CAPTURE_FINALIZER=${{ matrix.pod-status-finalizer || 'false' }}> /tmp/argo.log 2>&1 &
+            POD_STATUS_CAPTURE_FINALIZER=true > /tmp/argo.log 2>&1 &
```

Reply:
I've added the fix above.
https://github.com/argoproj/argo-workflows/pull/12413/commits/e2b2de4ac9b034ce3485dd90cc232846ee17c9b8